entrypoint.sh

```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
+set -e
+
+# if CMD was empty or `listen`, do `tail -f /dev/null` to keep the container running
+if [ "$#" -eq 0 -o "$1" = 'listen' ]; then
+  set -- tail -f /dev/null
```

Comment:
what exactly is set doing here? I'm not very familiar with its usage.

Reply:
`set` sets positional parameters when not setting options. `--` after `set` tells that anything after `--` is not option.

Basically, this file does something like
```ruby
(ARGV.clear; ARGV.push('tail', '-f', '/dev/null')) if ARGV.empty? || ARGV[0] == 'listen'
ARGV.unshift('node') if ARGV[0] == 'run' || ARGV[0] == 'run-json'
```
to manipulate `ARGV`.